src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs

```diff
@@ -41,7 +42,20 @@ public static class LockFileUtils
                 cache: new LockFileBuilderCache());
         }
 
+        /// <summary>
+        /// Create a lock file target library for the given <paramref name="library"/>
+        /// </summary>
+        /// <param name="libraryDependency">The library dependency equivalent to <paramref name="library"/>. The library dependency contains metadata applicable during asset selection. Can be null.</param>
+        /// <param name="library">The lock file library, expected to be for the equivalent package as <paramref name="library"/> and <paramref name="package"/>. </param>
+        /// <param name="package">The local package info.</param>
+        /// <param name="targetGraph">The target graph for which the asset selection needs to happen.</param>
+        /// <param name="dependencyType">The resolved dependency type.</param>
+        /// <param name="targetFrameworkOverride">The original framework if the asset selection is happening for a fallback framework.</param>
+        /// <param name="dependencies">The dependencies of this package.</param>
+        /// <param name="cache">The lock file build cache.</param>
+        /// <returns>The LockFileTargetLibrary</returns>
         public static LockFileTargetLibrary CreateLockFileTargetLibrary(
+                LibraryDependency libraryDependency,
```

Comment:
This is a breaking change to a public API. Can you create an overload with the original signature that calls this one with `libraryDependency` set to `null`?

Reply:
Honestly, these are just overkills. 
This method shouldn't even be public in the first place. :(
I can add it, but code like this affects our velocity so much.

Reply:
I added it back and obsoleted it, indicating that this method shouldn't be used, and to use the lock file builder instead.